c/bpxskt.c

```diff
@@ -369,18 +369,39 @@ Socket *tcpClient3(SocketAddress *socketAddress,
       /* manual says returnCode and value only meaningful if return value = -1 */
       *returnCode = 0;
       *reasonCode = 0;
+      if (readTimeoutInMillis >= 0) {
+        setSocketOption(socket, SOL_SOCKET, SOCK_SO_RCVTIMEO, sizeof(int), (char*)&readTimeoutInMillis, returnCode, reasonCode);
```

Comment:
`SOCK_SO_RCVTIMEO` accepts the `timeval` struct which is 8 bytes. Are we sure this actually works? Does the call return 0?

Reply:
I've performed a stand-alone test where I passed 4 bytes to `BPX1OPT` (this is what `setSocketOption` uses) and got `rv=-1, rc=121, rsn=0x11107315` which is:
```
JRInvalidOptLength: The option length specified is not valid

Action: The value specified for the option length is not valid. Re-issue the
socket call with a valid option length.
```